tools/run_tests/xds_k8s_test_driver/framework/infrastructure/traffic_director.py

```diff
@@ -122,13 +131,16 @@ def setup_routing_rule_map_for_grpc(self, service_host, service_port):
     def cleanup(self, *, force=False):
         # Cleanup in the reverse order of creation
         self.delete_forwarding_rule(force=force)
-        if self.target_proxy_is_http:
-            self.delete_target_http_proxy(force=force)
-        else:
-            self.delete_target_grpc_proxy(force=force)
+        # Variable target_proxy_is_http is only set in create_backend_service
+        # (otherwise it's always false), which means this will always try to
+        # delete grpc_proxy, strategy=keep will be stuck in deleting url-maps,
+        # backend services and others.
```

Comment:
Good finding.

Reply:
This comment is actually confusing (it explain the diff, but not the new code). I deleted it.